src/Microsoft.ML.Dnn/DnnRetrainTransform.cs

```diff
@@ -912,6 +915,8 @@ private void UpdateCacheIfNeeded(long position, ITensorValueGetter[] srcTensorGe
             {
                 if (outputCache.Position != position)
                 {
+                    if (_parent.Graph.graph_key != tf.get_default_graph().graph_key)
+                        _parent._session.graph.as_default();
```

Comment:
What do these lines do?

Reply:
if the current graph is not the tf default graph, this line sets it as the default graph.

---
In reply to: [324450509](https://github.com/dotnet/machinelearning/pull/4205#discussion_r324450509) [](ancestors = 324450509)